.github/workflows/flakiness-detection.yml

```diff
@@ -0,0 +1,97 @@
+      with:
+        go-version: 1.16
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Run setup
+      run: make install-tools tidy generate-mocks
+    - name: Run tests
+      run: make flakiness-run
+    - name: Set file name
+      run: echo "FLAKINESS_RUN_FILENAME=$GITHUB_SHA-$(git show --no-patch --no-notes --pretty='%ct' $GITHUB_SHA)-unit" >> $GITHUB_ENV
+    - id: Upload file
+      uses: google-github-actions/upload-cloud-storage@main
+      with:
+        credentials: ${{ secrets.gcp_credentials }}
+        path: ./flakiness_results
+        destination: flakiness-detection/${{ env.FLAKINESS_RUN_FILENAME }}
+
+  unit-test-crypto:
+    name: Unit Tests (crypto)
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Run tests
+      run: make -C crypto flakiness-run
+    - name: Set file name
+      run: echo "FLAKINESS_RUN_FILENAME=$GITHUB_SHA-$(git show --no-patch --no-notes --pretty='%ct' $GITHUB_SHA)-unit-crypto" >> $GITHUB_ENV
+    - id: Upload file
+      uses: google-github-actions/upload-cloud-storage@main
+      with:
+        credentials: ${{ secrets.gcp_credentials }}
+        path: ./crypto/flakiness_results
+        destination: flakiness-detection/${{ env.FLAKINESS_RUN_FILENAME }}
+
+  unit-test-integration:
+    name: Unit Tests (integration)
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Run tests
+      run: make -C integration flakiness-run
+    - name: Set file name
+      run: echo "FLAKINESS_RUN_FILENAME=$GITHUB_SHA-$(git show --no-patch --no-notes --pretty='%ct' $GITHUB_SHA)-unit-integration" >> $GITHUB_ENV
+    - id: Upload file
+      uses: google-github-actions/upload-cloud-storage@main
+      with:
+        credentials: ${{ secrets.gcp_credentials }}
+        path: ./integration/flakiness_results
+        destination: flakiness-detection/${{ env.FLAKINESS_RUN_FILENAME }}
+
+  integration-test:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Build relic
+      run: make crypto/relic/build
+    - name: Docker build
+      run: make docker-build-flow
+    - name: Run tests
+      run: make -C integration flakiness-run-integration
+    - name: Set file name
+      run: echo "FLAKINESS_RUN_FILENAME=$GITHUB_SHA-$(git show --no-patch --no-notes --pretty='%ct' $GITHUB_SHA)-integration" >> $GITHUB_ENV
+    - id: Upload file
+      uses: google-github-actions/upload-cloud-storage@main
+      with:
+        credentials: ${{ secrets.gcp_credentials }}
+        path: ./integration/flakiness_results
```

Comment:
There's a data race to the `./integration/flakiness_results` file name here, could we make them distinct?

Reply:
Since these are separate jobs, they will actually run on separate VM's in Github Actions. So each job will have its own version of the file.